src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax;
+
+public class Main {
+    public static void main(String[] args) {
+        FigureSupplier figureSupplier = new FigureSupplier();
+        Figure[] figures = new Figure[6];
+
+        for (int i = 0; i < figures.length / 2; i++) {
```

Comment:
You can simplify the loop by using 'i < figures.length' and eliminate the second loop. This will work if you handle the default figure case within the first loop when 'i' reaches the middle of the array.